reVX/least_cost_xmission/cost_creator.py

```diff
@@ -0,0 +1,304 @@
+# -*- coding: utf-8 -*-
+"""
+Module to build and save least cost raster layers
+"""
+import logging
+import numpy as np
+from reVX.handlers.geotiff import Geotiff
+from reV.handlers.exclusions import ExclusionLayers
+
+from reVX.utilities.exclusions_converter import ExclusionsConverter
+from .config import XmissionConfig, WATER_NLCD_CODE, WATER_MULT, \
+    METERS_IN_MILE, NLCD_LAND_USE_CLASSES, HILL_MULT, MTN_MULT, HILL_SLOPE, \
+    MTN_SLOPE, CELL_SIZE
+
+logger = logging.getLogger(__name__)
+
+
+class XmissionCostCreator(ExclusionsConverter):
+    """
+    Class to create and save Transmission cost layers to a .h5 Exclusion file
+    - dist_to_coast (for valid sc_points)
+    - base_costs
+    - multiplier_*mw
+    - xmission_barrier
+    """
+    def __init__(self, iso_regions_f, land_use_classes, iso_mults,
+                 base_line_costs, iso_lookup, default_mults,
+                 excl_h5='/shared-projects/rev/exclusions/ATB_Exclusions.h5',
+                 slope_lyr='srtm_slope', nlcd_lyr='usa_mrlc_nlcd2011'):
+        """
+        Parameters
+        ----------
+        TODO
+
+        iso_regions_f : str
+            ISO regions raster file
+        nlcd_f : str
+            NLCD (land use) raster file
+        slope_f : str
+            Slope raster file
+        land_use_classes : dict
+            NLCD land use codes corresponding to use types
+        iso_mults : list of dict
+            Line cost multipliers for land use and terrain by ISO
+        base_line_costs : dict
+            Dict with per miles line costs for various capacities by ISO.
+        iso_lookup : dict
+            Keys are ISO names used in various dicts with keys being int codes
+            use in the iso_regions_f.
+        default_mults : dict
+            Default line cost multipliers for areas not covered by the ISO
+            regions.
+        """
+        logger.info('Loading all files')
+        with Geotiff(iso_regions_f) as gt:
+            self._iso_regions = gt.values[0]
+            self._profile = gt.profile
+
+        with ExclusionLayers(excl_h5) as el:
```

Comment:
Move these to the create_multipliers method, they aren't needed here

Reply:
done